pennylane/operation.py

```diff
@@ -263,6 +268,31 @@ def grad_recipe(self):
         """
         return self._grad_recipe
 
+    @property
+    def generator(self):
+        r"""Generator of the operation.
+
+        A length-2 list ``[generator, scaling_factor]``, where
+
+        * ``generator`` is an existing PennyLane
+          operation class or :math:`2\times 2` Hermitian array
+          that acts as the generator of the current operation
+
+        * ``scaling_factor`` represents a scaling factor applied
+          to the generator operation
+
+        For example, if :math:`U(\theta)=e^{i0.7\theta \sigma_x}`, then
+        :math:`\sigma_x`, with scaling factor :math:`s`, is the generator
+        of operator :math:`U(\theta)`:
+
+        .. code-block:: python
+
+            generator = [PauliX, 0.7]
+
+        Default is ``[None, 1]``, indicating the operation has no generator.
+        """
+        return [None, 1]
```

Comment:
Good point. I think the thing that makes the most sense here is a tuple - since all the class attributes of `Operation` are developer facing, no need to overcomplicate it with a named tuple.

With respect to the `op.generator = 0.7 * PauliX`, I think my preference right now is to leave it as is. Mostly because this isn't user facing, so I'm not sure it matters having a slightly nicer UI. Plus, it avoids adding complex operator overloading for a very minor use case.

Reply:
I would use a named tuple _because_ it is developer-facing. In a more structured programming language it would use a struct here because I think that `generator.operator` and `generator.scaling` is much more readable than `generator[0]` and `generator[1]`.

In user facing code I would actually not use it because it is just not idiomatic in python.

Reply:
That's a good point. I would prefer for now to leave it as a list, just to match all the other class attributes (currently, we use lists for everything - the gradient recipes, heisenberg reps, etc.).

Perhaps we should have an issue/PR to explore alternatives such as named tuples.